### DISCUSSION TITLE
Zigbee <-> Matter bridge

### DISCUSSION THREAD

**Discussion by mszczepanczyk** [arendst/Tasmota#19435] [Q&A]
I'm trying to create sort of a bridge to switch lights from my phone via Tasmota. The flow looks like this:

`iPhone  <--Matter-->  Tasmota  <--Zigbee-->  Light switch`

I compiled Tasmota with `USE_MATTER_DEVICE` and put it on a Sonoff Zigbee Bridge Pro. Both protocols seem to work. I can toggle [an Ikea Tradfri on/off switch](https://zigbee.blakadder.com/Ikea_E1603.html) through console (e.g. `zbsend {"device":"0xAB35","send":{"power": 1}}`). I can also create Matter devices and see them on my iPhone (although they are "unresponsive"). 

The problem is that I can't seem to find a way to build a connection between these two protocols. I'm prepared to write some code so I think I'd get by with reading/writing on Matter devices (similar to `ZbInfo`/`ZbSend` with Zigbee) but I couldn't find any related documentation. I'm stuck on reading the implementation atm (`lib/libesp32/berry_matter/src/embedded`).

**Comment by sfromis** [comment]
What I'd try would be to have Matter control "virtual" devices, and then set up automation (like using Berry or rules) to replicate actions to/from Zigbee. Simple case of a virtual device could be to config a relay on an unused gpio, which is then made available via Matter.

**Comment by Jason2866** [comment]
A Feature will be added "Matter Event"
Like zigbee, it publishes mqtt events and json for rules whenever it receives a command

First PR https://github.com/arendst/Tasmota/pull/19484

**Reply by s-hadinger** [reply]
The necessary pieces are coming shortly.
